cli/command/service/update.go

```diff
@@ -433,6 +434,41 @@ func getUpdatedSecrets(apiClient client.APIClient, flags *pflag.FlagSet, secrets
 		}
 		secrets = append(secrets, addSecrets...)
 	}
+
+	// If we're being asked to replace a specific secret, let's do in-place replacement of names and IDs
+	if flags.Changed(flagSecretReplace) {
+		secretRefQuery := []*types.SecretRequestOption{}
+
+		// Retrieve all of the secret1=secret2 replacement mappings and create a list of SecretRequestOption
+		replaceValues := flags.Lookup(flagSecretReplace).Value.(*opts.MapOpts).GetAll()
+		for _, value := range replaceValues {
+			secretRefQuery = append(secretRefQuery, &types.SecretRequestOption{Source: value})
+		}
+
+		// Retrieve a list of secret references with the SecretID filled in
+		fullSecretRefs, err := parseSecrets(apiClient, secretRefQuery)
+		if err != nil {
+			return nil, err
+		}
+
+		// Create a map with all of the full secretReferences
+		replSecretMap := make(map[string]*swarm.SecretReference)
+		for _, secretRef := range fullSecretRefs {
+			replSecretMap[secretRef.SecretName] = secretRef
+		}
+
+		// Go through all of the secrets we wish to add and see if we're being asked to replace it
+		for _, secret := range secrets {
+			for original, new := range replaceValues {
```

Comment:
Since this is a map, rather than ranging over it, can you just index into it?  E.g.:
```
for _, secret := range secrets {
    if newSecretName, ok := replaceValues[secret.SecretName] {
        secret.SecretName = newSecretName
        secret.SecretID = replSecretMap[newSecretName].SecretID
    }
}
```